spec/api-deprecations-spec.js

```diff
@@ -18,9 +18,52 @@ describe('deprecations', () => {
     assert.deepEqual(messages, ['this is deprecated'])
   })
 
+  it('returns a deprecation handler after one is set', () => {
+    const messages = []
+
+    deprecations.setHandler((message) => {
+      messages.push(message)
+    })
+
+    deprecate.log('this is deprecated')
+    assert(typeof deprecations.getHandler() === 'function')
+  })
+
+  it('returns a deprecation warning', () => {
+    const messages = []
+
+    deprecations.setHandler((message) => {
+      messages.push(message)
+    })
+
+    deprecate.warn('old', 'new')
+    assert.deepEqual(messages, [`'old' is deprecated. Use 'new' instead.`])
```

Comment:
This is OK but a little brittle, it'll fail if we ever change the warning format. What would you think about something like

```
let warnings = 0
deprecations.setHandler(message => ++warnings)
deprecate.warn('old', 'new')
assert.equal(warnings, 1)
```